tests/unit/EditShareStructure.spec.ts

```diff
@@ -147,11 +147,7 @@ describe('Edit Share Structure component', () => {
   it('Emits add edit class event', async () => {
     const shareClass = createShareStructure(null, 1, 'Class', 'Class A', true, 100, true, 0.50, 'CAD', true)
     const wrapper: Wrapper<EditShareStructure> = createComponent(shareClass, -1, 1, null, [])
-    await Vue.nextTick()
```

Comment:
Removing it (same as emitting a remove class event), fixed the tests. I also reverted the optional chaining.

Reply:
That makes no sense to me. Probably it's one of those tests that doesn't check the right thing (for example, testing that something _doesn't_ exist, which passes when the code is broken).

I'm happy to ignore this for now if you want. Or if it really bugs you then I'll help you solve it, too :)

Reply:
You should be able to have "await next tick" everywhere with zero impact -- it would be the same as waiting for the user to click something.

If you want to try and fix this, revert the "await next tick" and look into the emitted events, etc. I dunno.

Reply:
If you don't await then maybe you're "expecting" results before the results are ready.

Reply:
That makes sense. I also was surprised that removing the "await next tick" actually fixed the tests. I'm just going to revert this change (again) and merge without fixing the tests if that's OK with you. This is outside the ticket's scope I'm working on anyway 😟

Reply:
LOL. Totally OK with me. Been there, done that, got the headache, took the Tylenol, lived to code another day.

Reply:
Hahahaha. I FEEL YOU ON THAT SEV.